--- git status ---
HEAD detached from dc9bbf2
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/actuators/actuator_hydraulic.py

no changes added to commit (use "git add" and/or "git commit -a") 


--- git diff ---
diff --git a/source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/actuators/actuator_hydraulic.py b/source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/actuators/actuator_hydraulic.py
index 4c97758..ca099e5 100644
--- a/source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/actuators/actuator_hydraulic.py
+++ b/source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/actuators/actuator_hydraulic.py
@@ -42,7 +42,7 @@ class IdentifiedActuatorHydraulic(DCMotor):
 
 
         # apply first order delay on the torque
-        control_action.joint_efforts = control_action.joint_efforts*self.first_order_delay_filter + \
+        """control_action.joint_efforts = control_action.joint_efforts*self.first_order_delay_filter + \
             (1.-self.first_order_delay_filter)*self.last_joint_efforts
         self.last_joint_efforts = control_action.joint_efforts
 
@@ -50,7 +50,7 @@ class IdentifiedActuatorHydraulic(DCMotor):
         # apply second order delay on the torque
         control_action.joint_efforts = control_action.joint_efforts*self.second_order_delay_filter + \
             (1.-self.second_order_delay_filter)*self.last_two_joint_efforts
-        self.last_two_joint_efforts = control_action.joint_efforts
+        self.last_two_joint_efforts = control_action.joint_efforts"""
 
 
         # apply friction model on the torque