--- git status ---
HEAD detached from 8be4db9
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   scripts/rsl_rl/play_symm.py
	modified:   scripts/sim_to_others/locomotion_policy_wrapper.py
	modified:   scripts/sim_to_others/play_mujoco.py
	modified:   source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/assets/aliengo_asset/from_urdf/Props/instanceable_meshes.usd
	modified:   source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/assets/aliengo_asset/from_urdf/aliengo.usd
	modified:   source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/assets/aliengo_asset/from_xml/Props/instanceable_meshes.usd
	modified:   source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/assets/aliengo_asset/from_xml/aliengo.usd
	modified:   source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/tasks/locomotion_old/aliengo_env_cfg.py

Untracked files:
  (use "git add <file>..." to include in what will be committed)
	tested_policies/aliengo/data_aug_policy/

no changes added to commit (use "git add" and/or "git commit -a") 


--- git diff ---
diff --git a/scripts/rsl_rl/play_symm.py b/scripts/rsl_rl/play_symm.py
index 9c1bf4e..9eb7b1f 100644
--- a/scripts/rsl_rl/play_symm.py
+++ b/scripts/rsl_rl/play_symm.py
@@ -38,7 +38,7 @@ import torch
 
 
 #from rsl_rl.runners import on_policy_runner
-from morphosymm_rl.on_policy_runner_symm import OnPolicyRunnerSymm as OnPolicyRunner
+from morphosymm_rl.symm_on_policy_runner import SymmOnPolicyRunner as OnPolicyRunner
 
 
 from isaaclab.envs import DirectMARLEnv, multi_agent_to_single_agent
diff --git a/scripts/sim_to_others/locomotion_policy_wrapper.py b/scripts/sim_to_others/locomotion_policy_wrapper.py
index 1f442be..62b3f52 100644
--- a/scripts/sim_to_others/locomotion_policy_wrapper.py
+++ b/scripts/sim_to_others/locomotion_policy_wrapper.py
@@ -21,10 +21,9 @@ from gym_quadruped.utils.quadruped_utils import LegsAttr
 
 import onnxruntime as ort
 
-policy_path = "/home/alienware/isaaclab_ws_home/quadruped_rl_collection/tested_policies/aliengo/equivariant_policy"
+policy_path = "/home/alienware/isaaclab_ws_home/basic-locomotion-dls-isaaclab/logs/rsl_rl/aliengo_rough_direct/2025-06-26_17-24-09"
 #policy_path = dir_path + "/../../tested_policies/aliengo/data_augment"
 policy_path = policy_path + "/exported/policy.onnx"
-policy_path = "/home/alienware/isaaclab_ws_home/quadruped_rl_collection/logs/rsl_rl/aliengo_rough_direct/2025-06-19_15-43-18/policy_7999.onnx"
 policy = ort.InferenceSession(policy_path)
 
 import torch
diff --git a/scripts/sim_to_others/play_mujoco.py b/scripts/sim_to_others/play_mujoco.py
index 1a59576..daca64b 100644
--- a/scripts/sim_to_others/play_mujoco.py
+++ b/scripts/sim_to_others/play_mujoco.py
@@ -26,7 +26,7 @@ if __name__ == '__main__':
 
     robot_name = "aliengo"
     robot_feet_geom_names = dict(FL='FL', FR='FR', RL='RL', RR='RR')
-    scene_name = "random_pyramids" #random_boxes
+    scene_name = "random_boxes" #random_boxes
     simulation_dt = 0.002
 
 
diff --git a/source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/assets/aliengo_asset/from_urdf/Props/instanceable_meshes.usd b/source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/assets/aliengo_asset/from_urdf/Props/instanceable_meshes.usd
index c2feec8..546ae96 100644
Binary files a/source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/assets/aliengo_asset/from_urdf/Props/instanceable_meshes.usd and b/source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/assets/aliengo_asset/from_urdf/Props/instanceable_meshes.usd differ
diff --git a/source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/assets/aliengo_asset/from_urdf/aliengo.usd b/source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/assets/aliengo_asset/from_urdf/aliengo.usd
index 4a7793e..f066d0a 100644
Binary files a/source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/assets/aliengo_asset/from_urdf/aliengo.usd and b/source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/assets/aliengo_asset/from_urdf/aliengo.usd differ
diff --git a/source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/assets/aliengo_asset/from_xml/Props/instanceable_meshes.usd b/source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/assets/aliengo_asset/from_xml/Props/instanceable_meshes.usd
index f4bcf2a..6d5f4c3 100644
Binary files a/source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/assets/aliengo_asset/from_xml/Props/instanceable_meshes.usd and b/source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/assets/aliengo_asset/from_xml/Props/instanceable_meshes.usd differ
diff --git a/source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/assets/aliengo_asset/from_xml/aliengo.usd b/source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/assets/aliengo_asset/from_xml/aliengo.usd
index 08ab35c..c18884b 100644
Binary files a/source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/assets/aliengo_asset/from_xml/aliengo.usd and b/source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/assets/aliengo_asset/from_xml/aliengo.usd differ
diff --git a/source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/tasks/locomotion_old/aliengo_env_cfg.py b/source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/tasks/locomotion_old/aliengo_env_cfg.py
index 9a48801..ee04f30 100644
--- a/source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/tasks/locomotion_old/aliengo_env_cfg.py
+++ b/source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/tasks/locomotion_old/aliengo_env_cfg.py
@@ -325,11 +325,11 @@ class AliengoRoughBlindEnvCfg(AliengoFlatEnvCfg):
                 proportion=0.1, slope_range=(0.2, 0.4), platform_width=2.0, border_width=0.25
             ),
             "pyramid_stairs": terrain_gen.MeshPyramidStairsTerrainCfg(
-                proportion=0.2, step_height_range=(0.05, 0.15), step_width=0.3,
+                proportion=0.2, step_height_range=(0.05, 0.18), step_width=0.3,
                 platform_width=3.0, border_width=1.0, holes=False,
             ),
             "pyramid_stairs_inv": terrain_gen.MeshInvertedPyramidStairsTerrainCfg(
-                proportion=0.1, step_height_range=(0.05, 0.15), step_width=0.3,
+                proportion=0.1, step_height_range=(0.05, 0.18), step_width=0.3,
                 platform_width=3.0, border_width=1.0, holes=False,
             ),
         },