--- git status ---
HEAD detached from dc9bbf2
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/assets/go2_asset.py
	modified:   source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/tasks/locomotion/agents/rsl_rl_ppo_cfg.py
	modified:   source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/tasks/locomotion/go2_env_cfg.py

no changes added to commit (use "git add" and/or "git commit -a") 


--- git diff ---
diff --git a/source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/assets/go2_asset.py b/source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/assets/go2_asset.py
index 7bff2e8..39a6f09 100644
--- a/source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/assets/go2_asset.py
+++ b/source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/assets/go2_asset.py
@@ -20,9 +20,9 @@ armature_mujoco = 0.01
 
 GO2_HIP_ACTUATOR_CFG = IdentifiedActuatorCfg(
     joint_names_expr=[".*_hip_joint"],
-    effort_limit=44.4,
-    velocity_limit=21.0,
-    saturation_effort=44.4,
+    effort_limit=23.7,
+    velocity_limit=30.1,
+    saturation_effort=23.7,
     stiffness=stiffness_mujoco,
     damping=damping_mujoco,
     armature=armature_mujoco,
@@ -33,9 +33,9 @@ GO2_HIP_ACTUATOR_CFG = IdentifiedActuatorCfg(
 
 GO2_THIGH_ACTUATOR_CFG = IdentifiedActuatorCfg(
     joint_names_expr=[".*_thigh_joint"],
-    effort_limit=44.4,
-    velocity_limit=21.0,
-    saturation_effort=44.4,
+    effort_limit=23.7,
+    velocity_limit=30.1,
+    saturation_effort=23.7,
     stiffness=stiffness_mujoco,
     damping=damping_mujoco,
     armature=armature_mujoco,
@@ -46,9 +46,9 @@ GO2_THIGH_ACTUATOR_CFG = IdentifiedActuatorCfg(
 
 GO2_CALF_ACTUATOR_CFG = IdentifiedActuatorCfg(
     joint_names_expr=[".*_calf_joint"],
-    effort_limit=44.4,
-    velocity_limit=21.0,
-    saturation_effort=44.4,
+    effort_limit=45.43,
+    velocity_limit=15.7,
+    saturation_effort=45.43,
     stiffness=stiffness_mujoco,
     damping=damping_mujoco,
     armature=armature_mujoco,
@@ -87,17 +87,6 @@ GO2_CFG = ArticulationCfg(
         joint_vel={".*": 0.0},
     ),
 
-    #actuators={
-    #    "base_legs": DCMotorCfg(
-    #        joint_names_expr=[".*_hip_joint", ".*_thigh_joint", ".*_calf_joint"],
-    #        effort_limit=44.4,
-    #        saturation_effort=44.4,
-    #        velocity_limit=21.0,
-    #        stiffness=25.0,
-    #        damping=2,
-    #        friction=0.0,
-    #    ),
-    #},
     
     actuators={"hip": GO2_HIP_ACTUATOR_CFG, "thigh": GO2_THIGH_ACTUATOR_CFG,
                "calf": GO2_CALF_ACTUATOR_CFG},
diff --git a/source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/tasks/locomotion/agents/rsl_rl_ppo_cfg.py b/source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/tasks/locomotion/agents/rsl_rl_ppo_cfg.py
index 0da22b7..e26db6b 100644
--- a/source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/tasks/locomotion/agents/rsl_rl_ppo_cfg.py
+++ b/source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/tasks/locomotion/agents/rsl_rl_ppo_cfg.py
@@ -129,7 +129,7 @@ class RoughPPORunnerCfg(RslRlOnPolicyRunnerCfg):
         activation="elu",
     )
     algorithm = RslRlPpoAlgorithmCfg(
-        class_name="PPO", #PPO, PPOSymmDataAugmented #AMP_PPO
+        class_name="PPOSymmDataAugmented", #PPO, PPOSymmDataAugmented #AMP_PPO
         value_loss_coef=1.0,
         use_clipped_value_loss=True,
         clip_param=0.2,
diff --git a/source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/tasks/locomotion/go2_env_cfg.py b/source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/tasks/locomotion/go2_env_cfg.py
index 3ad9c23..20c73b8 100644
--- a/source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/tasks/locomotion/go2_env_cfg.py
+++ b/source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/tasks/locomotion/go2_env_cfg.py
@@ -286,7 +286,7 @@ class Go2FlatEnvCfg(DirectRLEnvCfg):
     desired_gait = "trot" #crawl, pace, multigait
 
     # Desired tracking variables
-    desired_base_height = 0.27
+    desired_base_height = 0.30
     desired_feet_height = 0.05
 
     # Desired clip actions
@@ -349,7 +349,7 @@ class Go2RoughBlindEnvCfg(Go2FlatEnvCfg):
                 proportion=0.1, grid_width=0.45, grid_height_range=(0.05, 0.10), platform_width=2.0,
             ),
             "star": terrain_gen.MeshStarTerrainCfg(
-                proportion=0.1, num_bars=10, bar_width_range=(0.15, 0.20), bar_height_range=(0.05, 0.15), platform_width=2.0,
+                proportion=0.1, num_bars=10, bar_width_range=(0.15, 0.20), bar_height_range=(0.05, 0.13), platform_width=2.0,
             ),
             "random_rough": terrain_gen.HfRandomUniformTerrainCfg(
                 proportion=0.1, noise_range=(0.02, 0.06), noise_step=0.02, border_width=0.25
@@ -361,11 +361,11 @@ class Go2RoughBlindEnvCfg(Go2FlatEnvCfg):
                 proportion=0.1, slope_range=(0.2, 0.4), platform_width=2.0, border_width=0.25
             ),
             "pyramid_stairs": terrain_gen.MeshPyramidStairsTerrainCfg(
-                proportion=0.15, step_height_range=(0.05, 0.18), step_width=0.3,
+                proportion=0.15, step_height_range=(0.05, 0.13), step_width=0.3,
                 platform_width=3.0, border_width=1.0, holes=False,
             ),
             "pyramid_stairs_inv": terrain_gen.MeshInvertedPyramidStairsTerrainCfg(
-                proportion=0.15, step_height_range=(0.05, 0.18), step_width=0.3,
+                proportion=0.15, step_height_range=(0.05, 0.13), step_width=0.3,
                 platform_width=3.0, border_width=1.0, holes=False,
             ),
         },