--- git status ---
HEAD detached from dc9bbf2
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/assets/hyqreal_asset.py
	modified:   source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/tasks/locomotion/agents/rsl_rl_ppo_cfg.py
	modified:   source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/tasks/locomotion/hyqreal_env_cfg.py

Untracked files:
  (use "git add <file>..." to include in what will be committed)
	tested_policies/hyqreal2/2025-09-03_15-21-35_8k_128_128_128_hyqreal_second_order_delay/

no changes added to commit (use "git add" and/or "git commit -a") 


--- git diff ---
diff --git a/source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/assets/hyqreal_asset.py b/source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/assets/hyqreal_asset.py
index 003d6cb..7ac06a6 100644
--- a/source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/assets/hyqreal_asset.py
+++ b/source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/assets/hyqreal_asset.py
@@ -13,12 +13,26 @@ from basic_locomotion_dls_isaaclab.assets import ISAAC_ASSET_DIR
 # HYQREAL robot configuration from mujoco
 stiffness_mujoco = 300.0 #200.0
 damping_mujoco = 20.0 #20.0
-friction_static_mujoco = 0.2
-friction_dynamic_mujoco = 0.6
 armature_mujoco = 0.01
 first_order_delay_filter = 0.1
 second_order_delay_filter = 0.1
 
+friction_static_mujoco = 0.2 * 0.0
+friction_dynamic_mujoco = 0.6 * 0.0
+activation_vel = 0.1# * 0.0
+
+static_friction_hip = 7.0 
+dynamic_friction_hip = 0.3
+viscous_friction_hip = 0.3
+
+static_friction_thigh = 7.0
+dynamic_friction_thigh = 0.3
+viscous_friction_thigh = 0.3
+
+static_friction_calf = 0.5
+dynamic_friction_calf = 0.3
+viscous_friction_calf = 0.3
+
 HYQREAL_HIP_ACTUATOR_CFG = IdentifiedActuatorHydraulicCfg(
     joint_names_expr=[".*_hip_joint"],
     effort_limit=173.0,
@@ -28,10 +42,14 @@ HYQREAL_HIP_ACTUATOR_CFG = IdentifiedActuatorHydraulicCfg(
     damping=damping_mujoco,
     armature=armature_mujoco,
     friction_static=friction_static_mujoco,
-    activation_vel=0.1,
+    activation_vel=activation_vel,
     friction_dynamic=friction_dynamic_mujoco,
     first_order_delay_filter=first_order_delay_filter,
     second_order_delay_filter=second_order_delay_filter,
+
+    friction = static_friction_hip,
+    dynamic_friction = dynamic_friction_hip,
+    viscous_friction = viscous_friction_hip,
 )
 
 HYQREAL_THIGH_ACTUATOR_CFG = IdentifiedActuatorHydraulicCfg(
@@ -47,6 +65,10 @@ HYQREAL_THIGH_ACTUATOR_CFG = IdentifiedActuatorHydraulicCfg(
     friction_dynamic=friction_dynamic_mujoco,
     first_order_delay_filter=first_order_delay_filter,
     second_order_delay_filter=second_order_delay_filter,
+
+    friction = static_friction_thigh,
+    dynamic_friction = dynamic_friction_thigh,
+    viscous_friction = viscous_friction_thigh,
 )
 
 HYQREAL_CALF_ACTUATOR_CFG = IdentifiedActuatorHydraulicCfg(
@@ -62,6 +84,10 @@ HYQREAL_CALF_ACTUATOR_CFG = IdentifiedActuatorHydraulicCfg(
     friction_dynamic=friction_dynamic_mujoco,
     first_order_delay_filter=first_order_delay_filter,
     second_order_delay_filter=second_order_delay_filter,
+
+    friction = static_friction_calf,
+    dynamic_friction = dynamic_friction_calf,
+    viscous_friction = viscous_friction_calf,
 )
 
 
diff --git a/source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/tasks/locomotion/agents/rsl_rl_ppo_cfg.py b/source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/tasks/locomotion/agents/rsl_rl_ppo_cfg.py
index ed6db40..82aba1f 100644
--- a/source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/tasks/locomotion/agents/rsl_rl_ppo_cfg.py
+++ b/source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/tasks/locomotion/agents/rsl_rl_ppo_cfg.py
@@ -137,7 +137,7 @@ class RoughPPORunnerCfg(RslRlOnPolicyRunnerCfg):
         activation="elu",
     )
     algorithm = RslRlPpoAlgorithmCfg(
-        class_name="PPO", #PPO, PPOSymmDataAugmented #AMP_PPO
+        class_name="PPOSymmDataAugmented", #PPO, PPOSymmDataAugmented #AMP_PPO
         value_loss_coef=1.0,
         use_clipped_value_loss=True,
         clip_param=0.2,
diff --git a/source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/tasks/locomotion/hyqreal_env_cfg.py b/source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/tasks/locomotion/hyqreal_env_cfg.py
index df16364..3d33e76 100644
--- a/source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/tasks/locomotion/hyqreal_env_cfg.py
+++ b/source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/tasks/locomotion/hyqreal_env_cfg.py
@@ -64,7 +64,7 @@ class EventCfg:
     )
     
 
-    scale_all_joint_friction_model = EventTerm(
+    """scale_all_joint_friction_model = EventTerm(
         func=custom_events.randomize_joint_friction_model,
         mode="startup",
         params={"asset_cfg": SceneEntityCfg("robot", joint_names=[".*"]), 
@@ -79,6 +79,18 @@ class EventCfg:
         params={"asset_cfg": SceneEntityCfg("robot", joint_names=[".*"]), 
                 "armature_distribution_params": (0.0, 1.0),
                 "operation": "scale"},
+    )"""
+
+    randomize_joint_parameters = EventTerm(
+        func=mdp.randomize_joint_parameters,
+        mode="reset",
+        params={
+            "asset_cfg": SceneEntityCfg("robot", joint_names=[".*"]), 
+            "friction_distribution_params": (0.2, 2.0),
+            "armature_distribution_params": (0.0, 1.0),
+            "operation": "scale",
+            "distribution": "uniform",
+        },
     )
 
     scale_all_joint_first_order_delay_filter_model = EventTerm(
@@ -103,8 +115,8 @@ class EventCfg:
     mode="reset",
     params={
         "asset_cfg": SceneEntityCfg("robot", joint_names=".*"),
-        "stiffness_distribution_params": (-15.0, 15.0),
-        "damping_distribution_params": (-3.0, 3.0),
+        "stiffness_distribution_params": (-20.0, 20.0),
+        "damping_distribution_params": (-5.0, 5.0),
         "operation": "add",
         "distribution": "uniform",
     },