--- git status ---
HEAD detached from dc9bbf2
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/assets/go2_asset.py
	modified:   source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/tasks/locomotion/agents/rsl_rl_ppo_cfg.py
	modified:   source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/tasks/locomotion/go2_env_cfg.py
	modified:   source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/tasks/locomotion/locomotion_env.py

Untracked files:
  (use "git add <file>..." to include in what will be committed)
	tested_policies/hyqreal2/2025-09-03_15-21-35_8k_128_128_128_hyqreal_second_order_delay/
	tested_policies/hyqreal2/2025-09-06_09-30-01_8k_128_128_128_hyqreal_second_order_delay_new_friction/

no changes added to commit (use "git add" and/or "git commit -a") 


--- git diff ---
diff --git a/source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/assets/go2_asset.py b/source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/assets/go2_asset.py
index 9f4a748..6d97003 100644
--- a/source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/assets/go2_asset.py
+++ b/source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/assets/go2_asset.py
@@ -13,10 +13,25 @@ from basic_locomotion_dls_isaaclab.assets import ISAAC_ASSET_DIR
 # Aliengo robot configuration from mujoco
 stiffness_mujoco = 25.0
 damping_mujoco = 2.0
-friction_static_mujoco = 0.2
-friction_dynamic_mujoco = 0.6
+
 armature_mujoco = 0.01
 
+friction_static_mujoco = 0.2 * 0.0
+friction_dynamic_mujoco = 0.6 * 0.0
+activation_vel = 0.1# * 0.0
+
+static_friction_hip = 0.5
+dynamic_friction_hip = 0.3
+viscous_friction_hip = 0.3
+
+static_friction_thigh = 0.5
+dynamic_friction_thigh = 0.3
+viscous_friction_thigh = 0.3
+
+static_friction_calf = 0.5
+dynamic_friction_calf = 0.3
+viscous_friction_calf = 0.3
+
 GO2_HIP_ACTUATOR_CFG = IdentifiedActuatorElectricCfg(
     joint_names_expr=[".*_hip_joint"],
     effort_limit=23.7,
@@ -26,8 +41,12 @@ GO2_HIP_ACTUATOR_CFG = IdentifiedActuatorElectricCfg(
     damping=damping_mujoco,
     armature=armature_mujoco,
     friction_static=friction_static_mujoco,
-    activation_vel=0.1,
+    activation_vel=activation_vel,
     friction_dynamic=friction_dynamic_mujoco,
+
+    friction = static_friction_hip,
+    dynamic_friction = dynamic_friction_hip,
+    viscous_friction = viscous_friction_hip,
 )
 
 GO2_THIGH_ACTUATOR_CFG = IdentifiedActuatorElectricCfg(
@@ -39,8 +58,12 @@ GO2_THIGH_ACTUATOR_CFG = IdentifiedActuatorElectricCfg(
     damping=damping_mujoco,
     armature=armature_mujoco,
     friction_static=friction_static_mujoco,
-    activation_vel=0.1,
+    activation_vel=activation_vel,
     friction_dynamic=friction_dynamic_mujoco,
+
+    friction = static_friction_thigh,
+    dynamic_friction = dynamic_friction_thigh,
+    viscous_friction = viscous_friction_thigh,
 )
 
 GO2_CALF_ACTUATOR_CFG = IdentifiedActuatorElectricCfg(
@@ -52,8 +75,12 @@ GO2_CALF_ACTUATOR_CFG = IdentifiedActuatorElectricCfg(
     damping=damping_mujoco,
     armature=armature_mujoco,
     friction_static=friction_static_mujoco,
-    activation_vel=0.1,
+    activation_vel=activation_vel,
     friction_dynamic=friction_dynamic_mujoco,
+
+    friction = static_friction_calf,
+    dynamic_friction = dynamic_friction_calf,
+    viscous_friction = viscous_friction_calf,
 )
 
 
diff --git a/source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/tasks/locomotion/agents/rsl_rl_ppo_cfg.py b/source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/tasks/locomotion/agents/rsl_rl_ppo_cfg.py
index ed6db40..82aba1f 100644
--- a/source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/tasks/locomotion/agents/rsl_rl_ppo_cfg.py
+++ b/source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/tasks/locomotion/agents/rsl_rl_ppo_cfg.py
@@ -137,7 +137,7 @@ class RoughPPORunnerCfg(RslRlOnPolicyRunnerCfg):
         activation="elu",
     )
     algorithm = RslRlPpoAlgorithmCfg(
-        class_name="PPO", #PPO, PPOSymmDataAugmented #AMP_PPO
+        class_name="PPOSymmDataAugmented", #PPO, PPOSymmDataAugmented #AMP_PPO
         value_loss_coef=1.0,
         use_clipped_value_loss=True,
         clip_param=0.2,
diff --git a/source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/tasks/locomotion/go2_env_cfg.py b/source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/tasks/locomotion/go2_env_cfg.py
index a698d24..8202da6 100644
--- a/source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/tasks/locomotion/go2_env_cfg.py
+++ b/source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/tasks/locomotion/go2_env_cfg.py
@@ -147,12 +147,12 @@ class Go2FlatEnvCfg(DirectRLEnvCfg):
 
     use_imu = False
 
-    use_cuncurrent_state_est = False
+    use_cuncurrent_state_est = True
     if(use_cuncurrent_state_est):
         cuncurrent_state_est_output_space = 3 #lin_vel_b
         single_cuncurrent_state_est_observation_space = single_observation_space
         cuncurrent_state_est_observation_space = observation_space
-        cuncurrent_state_est_batch_size = 32
+        cuncurrent_state_est_batch_size = 8
         cuncurrent_state_est_train_epochs = 500
         cuncurrent_state_est_lr = 1e-3
         cuncurrent_state_est_ep_saving_interval = 1000
@@ -236,8 +236,13 @@ class Go2FlatEnvCfg(DirectRLEnvCfg):
         mesh_prim_paths=["/World/ground"],
     )
 
-    # an imu sensor in case we don't want any state estimator
-    imu = ImuCfg(prim_path="/World/envs/env_.*/Robot/base", debug_vis=True)
+    # an imu sensor in case we don't want any state estimator (for now we can't use sites from the xml)
+    imu = ImuCfg(
+        prim_path="/World/envs/env_.*/Robot/base", 
+        offset=ImuCfg.OffsetCfg(
+            pos=(-0.02557, 0, 0.04232)
+        ), 
+        debug_vis=False)
 
 
     # scene
diff --git a/source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/tasks/locomotion/locomotion_env.py b/source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/tasks/locomotion/locomotion_env.py
index 77e87b5..63ef1d6 100644
--- a/source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/tasks/locomotion/locomotion_env.py
+++ b/source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/tasks/locomotion/locomotion_env.py
@@ -659,7 +659,7 @@ class LocomotionEnv(DirectRLEnv):
             obs_cuncurrent_state_est = self._observation_noise_model(obs_cuncurrent_state_est)   
 
         # Saving data
-        output_cuncurrent_state_est = torch.cat((self._robot.data.root_lin_vel_b), dim=-1)
+        output_cuncurrent_state_est = self._robot.data.root_lin_vel_b
         self._cuncurrent_state_est_network.dataset.add_sample(obs_cuncurrent_state_est, output_cuncurrent_state_est)
 
         # Prediction